.github/workflows/build-deps.yml

```diff
@@ -0,0 +1,196 @@
+name: Build emsdk dependencies
+
+on:
+  workflow_dispatch:
+    inputs:
+      llvm_build:
+        description: 'LLVM Webassembly arm64'
+        required: false
+        default: false
+        type: boolean
+      llvm_build_cross_compilation:
+        description: 'LLVM Webassembly arm64 (cross compilation)'
+        required: false
+        default: false
+        type: boolean
+      emscripten_build:
+        description: 'Emscripten'
+        required: false
+        default: false
+        type: boolean
+      nodejs_build:
+        description: 'Node.js arm64'
+        required: false
+        default: false
+        type: boolean
+      emsdk_build:
+        description: 'emsdk arm64'
+        required: false
+        default: false
+        type: boolean
+
+jobs:
+  build-llvm-webassembly-arm64:
+    if: ${{ inputs.llvm_build }} ||  ${{ inputs.emsdk_build }}
+
+    runs-on: [self-hosted, Windows, ARM64, WASM]
+    timeout-minutes: 600
+
+    steps:
+    
+    - name: Build LLVM WebAssembly arm64
+      shell: cmd
+      run: |
+        exit
+        git clone --config core.autocrlf=false https://github.com/llvm/llvm-project.git --single-branch --depth 1
+        cd llvm-project
+        if exist build_arm64 rmdir /s /q build_arm64
+        call "%ProgramFiles%\Microsoft Visual Studio\2022\Preview\VC\Auxiliary\Build\vcvarsall.bat" arm64
+        cmake -G Ninja -S llvm -B build_arm64 ^
+          -DCLANG_ENABLE_ARCMT=OFF ^
+          -DCLANG_ENABLE_STATIC_ANALYZER=OFF ^
+          -DCMAKE_BUILD_TYPE=Release ^
+          -DCMAKE_BUILD_WITH_INSTALL_RPATH=ON ^
+          -DCMAKE_CXX_COMPILER_LAUNCHER='ccache' ^
+          -DCMAKE_EXPORT_COMPILE_COMMANDS=ON ^
+          -DCMAKE_LINKER="C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\Llvm\x64\bin\lld-link.exe" ^
```

Comment:
Why this needs to be specified explicitly? Is it any known issue? If yes, can I have the link to the issue, please?

Reply:
It enables optimization and reduce llvm build a lot. only needed tools will be used for emsdk. It was done based on emscripten-release pipeline from here 
https://chromium.googlesource.com/emscripten-releases/+/refs/heads/main/src/build.py#681

Reply:
Sorry, for not being explicit. I was specifically referring to `-DCMAKE_LINKER="C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\Llvm\x64\bin\lld-link.exe"`  line?

Reply:
for some reason it is required by cmake

Reply:
Yeah, I believe, it's worth investigation what is that reason...

Reply:
Hmm, it might be related to the issue with 

```
"link.exe" "-out:C:\\Users\\radekbarton\\Projects\\emsdk\\emscripten\\main\\out\\test\\base64.cpp.native" -defaultlib:libcmt -defaultlib:oldnames -nologo "C:\\Users\\RADEKB~1\\AppData\\Local\\Temp\\base64-82da62.o"clang++: error: unable to execute command: program not executableclang++: error: linker command failed with exit code 1 (use -v to see invocation)
```

Reply:
If I remember correctly it was related to -DLLVM_TOOL_LTO_BUILD=OFF usage
I will double check it

Reply:
It will be investigated in next PR